engine/src/main/java/org/terasology/engine/entitySystem/DoNotPersist.java

```diff
@@ -0,0 +1,17 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.engine.entitySystem;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation used to mark components that are not to be stored upon serialization of an entity.
```

Comment:
Is this only about _serialization_ for persistence (e.g., save game) or also about _network serialization_?

Reply:
I would opt for a @Persisted flag then @DoNotPersist but I do realize that would be a very large change. When I think about serialization I would prefer to opt into persisting something to disk vs not persisting it.

Reply:
It is about serialization for persistence, the network serialization does not use the default component serialization check. But considering this I think it is more reasonable to reinstate the "null" component check and introduce a PersistenceComponentSerializeCheck that looks for the annotation. (I will do this on my local branch.)